src/routes/signup.ts

```diff
@@ -0,0 +1,53 @@
+import express from "express"
+import { DataResponse, SignupRequest } from "../interfaces/interfaces"
+
+import bcrypt from 'bcrypt';
+import jwt from 'jsonwebtoken';
+import { PrismaClient } from "@prisma/client";
+
+const router = express.Router()
+const prisma = new PrismaClient()
+
+router.post<{}, DataResponse, SignupRequest>('/signup', async (req, res, next) => {
+    const response: DataResponse = { data: null, errors: [] }
+    const { email, password } = req.body as SignupRequest
+
+    //TODO: make sure only @unal.edu.co emails are allowed
+    if (typeof email !== 'string' || typeof password !== 'string') {
+        response.errors.push({ message: 'Invalid user information has been provided' })
+        res.status(400).json(response)
+        return
+    }
+
+    const salt = await bcrypt.genSalt()
+    const hashedPassword = await bcrypt.hash(password, salt)
+
+    //TODO: improve error handling to inform that the email already has an account, if that's the case
+    const user = await prisma.user.create({
+        data: { email: email.trim(), password: hashedPassword },
```

Comment:
Can you make the email be in lower case too?
```suggestion
        data: { email: email.toLowerCase().trim(), password: hashedPassword },
```